moveit_ros/planning/planning_request_adapter_plugins/src/fix_start_state_collision.cpp

```diff
@@ -115,37 +127,106 @@ class FixStartStateCollision : public planning_request_adapter::PlanningRequestA
       planning_scene->checkCollision(vcreq, vcres, start_state);
 
       if (creq.group_name.empty())
-        ROS_INFO("Start state appears to be in collision");
+        ROS_INFO("Start state appears to be in collision. Attempting to fix.");
       else
-        ROS_INFO_STREAM("Start state appears to be in collision with respect to group " << creq.group_name);
+        ROS_INFO_STREAM("Start state appears to be in collision with respect to group " << creq.group_name
+                                                                                        << ". Attempting to fix.");
 
-      moveit::core::RobotStatePtr prefix_state(new moveit::core::RobotState(start_state));
-      random_numbers::RandomNumberGenerator& rng = prefix_state->getRandomNumberGenerator();
+      moveit::core::RobotStatePtr original_state(new moveit::core::RobotState(start_state));
+      random_numbers::RandomNumberGenerator& rng = original_state->getRandomNumberGenerator();
 
       const std::vector<const moveit::core::JointModel*>& jmodels =
           planning_scene->getRobotModel()->hasJointModelGroup(req.group_name) ?
               planning_scene->getRobotModel()->getJointModelGroup(req.group_name)->getJointModels() :
               planning_scene->getRobotModel()->getJointModels();
 
+      // Record the cartesian pose of every link to ensure no sudden movements will occur.
+      // The "furthest" links are checked first, as their movement is the largest.
+
+      // Check if an end effector link is set that is not part of the robot
+      const moveit::core::LinkModel* ee_link_model;
+      if (!req.goal_constraints.empty())
+      {
+        if (!req.goal_constraints[0].position_constraints.empty())
+          ee_link_model = start_state.getLinkModel(req.goal_constraints[0].position_constraints[0].link_name);
+        else if (!req.goal_constraints[0].orientation_constraints.empty())
+          ee_link_model = start_state.getLinkModel(req.goal_constraints[0].orientation_constraints[0].link_name);
+      }
+      else
+      {
+        ROS_ERROR("No goal constraint set; aborting FixStartStateCollision.");
```

Comment:
When would this happen? Not changing anything will make it fail at the end, right?